knora/dsplib/utils/xml_upload.py

```diff
@@ -221,6 +222,68 @@ def _parse_xml_file(input_file: str) -> etree.ElementTree:
     return tree
 
 
+def _perform_deep_inspection(
+    resources: list[XMLResource],
+    resclass_name_2_type: dict[str, type],
+    verbose: bool = False
+) -> None:
+    """
+    Inspects a list of resources more thoroughly than an XSD validation can do. Mainly, it checks if the resource types
+    and properties in the XML are consistent with the ontology.
+
+    Args:
+        resources: a list of parsed XMLResources
+        resclass_name_2_type: infos about the resource classes that exist on the DSP server for the current ontology
+        verbose: verbose switch
+
+    Returns:
+        None if everything went well. Raises a BaseError if there is a problem.
+    """
+    if verbose:
+        print("Perform a deep inspection of your XML file...")
+    knora_properties = resclass_name_2_type[resources[0].restype].knora_properties
```

Comment:
this throws a not-so-informative error if `resources` is empty. Can this ever happen? Then you might want to test it, and if needed provide a more informative error.

Reply:
no, it can't